cmd/nerdctl/run.go

```diff
@@ -635,13 +638,31 @@ func createContainer(cmd *cobra.Command, ctx context.Context, client *containerd
 	var s specs.Spec
 	spec := containerd.WithSpec(&s, opts...)
 	cOpts = append(cOpts, spec)
-
 	logrus.Debugf("final cOpts is %v", cOpts)
 	container, err := client.NewContainer(ctx, id, cOpts...)
 	if err != nil {
-		return nil, "", nil, err
+		gcContainer := func() {
+			var isErr bool
+			if errE := os.RemoveAll(stateDir); errE != nil {
+				isErr = true
```

Comment:
Should we return if stateDir deletion fails?

Reply:
this idea is to make generic msg  when it comes to cleanUp